internal/backend/runtime/omni/controllers/omni/cluster_machine_config.go

```diff
@@ -459,9 +459,13 @@ func buildInstallImage(imageFactoryHost string, resID resource.ID, installImage
 		return "", fmt.Errorf("machine %q has no secure boot status set", resID)
 	}
 
-	installerName := "installer"
+	if installImage.Platform == "" {
```

Comment:
if we want to avoid re-generating machine config, in theory we could do it only for Talos 1.10+

Reply:
or specifically populate `.platform` only for 1.10+

Reply:
changed the pr to do that